upgrade/scripts/upgrade/prerequisites.sh

```diff
@@ -1160,6 +1160,46 @@ if [[ ${state_recorded} == "0" && $(hostname) == "${PRIMARY_NODE}" ]]; then
       csi handoff bss-update-param --set "${bootparameter}"
     done
 
+    # Get a list of NCNs.
+    if IFS=$'\n' read -rd '' -a NCN_XNAMES; then
+      :
+    fi <<< "$(cray hsm state components list --role Management --type Node --format json | jq -r '.Components | map(.ID) | join("\n")')"
```

Comment:
We could theoretically filter on `--subrole Worker` here, but there's no harm in changing everything I guess.

Reply:
I assumed that param should not exist on all NCN's. I can switch it if needed / it may break something.

Reply:
To Don's point, `hsn` NICs only exist on workers so it's just less to process looking at. However, it doesn't hurt to scrub every NCN.